File: dailyQuestion/2020-07/07-02/golang/solution_baoli.go
package main

import (
	"fmt"
	"sort"
)

func kthSmallest(matrix [][]int, k int) int {
	p := []int{}
	n := len(matrix)
	for i := 0; i < n; i++ {
		p = append(p, matrix[i]...)
	}
	sort.Ints(p)
	return p[k-1]
}

func main() {
	matrix := [][]int{[]int{1, 5, 9}, []int{10, 11, 13}, []int{12, 13, 15}}
	result := kthSmallest(matrix, 8)
	fmt.Println(result)
}

File: dailyQuestion/2020/2020-07/07-02/python/solution_baoli.py
def kthSmallest(matrix: [[int]], k: int) -> int:
    p = sorted(sum(matrix, []))
    return p[k - 1]


if __name__ == "__main__":
    matrix = [[1, 5, 9],
              [10, 11, 13],
              [12, 13, 15]]
    result = kthSmallest(matrix, 8)
    print(result)
